sycl/include/sycl/ext/oneapi/matrix/matrix-unified.hpp

```diff
@@ -40,7 +40,8 @@ struct joint_matrix {
 
 #if defined(__SYCL_DEVICE_ONLY__)
 #if defined(__NVPTX__)
-  sycl::ext::oneapi::detail::joint_matrix_cuda<T, Use, Rows, Cols, Layout>
+  mutable sycl::ext::oneapi::detail::joint_matrix_cuda<T, Use, Rows, Cols,
```

Comment:
TODO: we should modify cuda's code since we add "const" qualifier to jm_store's jm. i added mutable here as a temp fix/
the mutable keyword is used to indicate that a particular member variable of a class is allowed to be modified even when the class object is considered const.
@JackAKirk , will you take care of cuda's changes  since "const" qualifier is added to jm_store's jm?

Reply:
Is this really required? the cuda implementation of `joint_matrix_store` does not modify the `joint_matrix` argument.
Can you give some more details why you think this change is required? Thanks

Reply:
actually, we can't pass const join_matrix's member join_matrix_cuda(it is also const ) to joint_matrix_store_cuda:
```
void joint_matrix_store_cuda(
    joint_matrix_cuda<
        T, sycl::ext::oneapi::experimental::matrix::use::accumulator, NumRows,
        NumCols, sycl::ext::oneapi::experimental::matrix::layout::dynamic> &src,
    multi_ptr<T, Space, IsDecorated> dst, size_t stride,
    sycl::ext::oneapi::experimental::matrix::layout Layout)
```
if i add "const " to joint_matrix_store_cuda's src, there is other issues too.

Reply:
I don't understand but I will check out this branch and see what errors there are.

Reply:
OK I understand the issue now. The fix is simple but requires quite a lot of repetitive code changes. I think it is fine to do as you suggest and use `mutable` temporarily.

Reply:
I can make a follow up PR to fix it correctly.

Reply:
On HIP it works without the need for mutable.
https://github.com/intel/llvm/pull/11485/commits/42e0c62ee83be46257f8f742ff9a97c5ef4e7a62

Reply:
> On HIP it works without the need for mutable. [42e0c62](https://github.com/intel/llvm/commit/42e0c62ee83be46257f8f742ff9a97c5ef4e7a62)

cuda requires reinterpret casts between signed / unsigned builtins for load/store, this is the origin of the problem. I don't think it affects AMD since amd  doesn't have builtins for load/store

Reply:
there might be some similarities, I think. Initially it worked only with mutable. A number of changes needed to make it work without mutable.

Reply:
oh yeah sure. I will make follow up changes for cuda after this is merged too.